test/functional/test-xhr.js

```diff
@@ -56,10 +58,7 @@ describe.configure().skipSafari().run('XHR', function() {
 
   function setupMockXhr() {
     const mockXhr = sandbox.useFakeXMLHttpRequest();
-    requests = [];
-    mockXhr.onCreate = function(xhr) {
-      requests.push(xhr);
-    };
+    xhrCreated = new Promise(resolve => mockXhr.onCreate = resolve);
```

Comment:
Cool, though I'm not sure how you'd verify multiple XHRs with this. No action required.

Reply:
Yeah this would only work for a single XHR, but the changes I made in `test-live-list-manager.js` is for multiple XHRs. I think writing it in the single XHR style is simpler and more readable, if the tests only need to verify a single XHR.